DevCycle/Networking/DevCycleService.swift

```diff
@@ -147,6 +151,43 @@ class DevCycleService: DevCycleServiceProtocol {
         }
     }
     
+    func saveEntity(user: DVCUser, completion: @escaping SaveEntityCompletionHandler) {
+        var saveEntityRequest = createSaveEntityRequest()
+        
+        guard let userIsAnonymous = user.isAnonymous, !userIsAnonymous else {
+            Log.error("Cannot save user data for an anonymous user!")
+            return
+        }
+        
+        let userEncoder = JSONEncoder()
+        userEncoder.dateEncodingStrategy = .iso8601
+        
+        guard let userData = try? userEncoder.encode(user) else {
+            return completion((nil, nil, ClientError.MissingUserOrFeatureVariationsMap))
+        }
+        
+        guard let userBody = try? JSONSerialization.jsonObject(with: userData, options: .fragmentsAllowed) else {
+            return completion((nil, nil, ClientError.MissingUserOrFeatureVariationsMap))
+        }
+        
+        let jsonBody = try? JSONSerialization.data(withJSONObject: userBody, options: .prettyPrinted)
+        
+        saveEntityRequest.httpMethod = "PATCH"
+        saveEntityRequest.addValue("application/json", forHTTPHeaderField: "Content-Type")
+        saveEntityRequest.addValue("application/json", forHTTPHeaderField: "Accept")
+        saveEntityRequest.addValue(config.environmentKey, forHTTPHeaderField: "Authorization")
+        Log.info("Save entity payload: \(String(data: jsonBody!, encoding: .utf8) ?? "")")
+        
+        saveEntityRequest.httpBody = jsonBody
+        
+        self.makeRequest(request: saveEntityRequest) { data, response, error in
+            if error != nil || data == nil {
```

Comment:
bit confused about this. Whats the point in checking if `data == nil` here? Wont that be the same outcome as just calling `completion((data, response, error))` since if that check is hit then we know that `error == nil`? I dont know ios so could be a dumb comment

Reply:
@jsalaber thoughts ^?

Reply:
looks like it was because we did operations based on if error is present or if data was nil, we'd log an error: https://github.com/DevCycleHQ/ios-client-sdk/commit/35c9a6237628452a8f165438e7b644ad46c27e03

 Since we don't, I think you can refactor to make this more sense.